ansible/library/console_facts.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+module:         console_facts
+version_added:  "2.0"
+author:         Jing Kan (jika@microsoft.com)
+short_description: Retrieve console information from Quagga
+description:
+    - Retrieve console feature information from CONFIG_DB
+    - Retrieve console status from show line command
+    - Retrieved facts will be inserted into the 'console_facts' key
+'''
+
+class ConsoleModule(object):
+    LINE_INDEX = 0
+    BAUD_INDEX = 1
+    FLCT_INDEX = 2
+    RDEV_INDEX = 5
+
+    STATUS_BUSY = "BUSY"
+    STATUS_IDLE = "IDLE"
+
+    STATUS_INDICATOR = "*"
+    FLCT_ENABLED_TEXT = "Enabled"
+
+    def __init__(self, include_remote_device_mapping = True):
+        self.include_remote_device_mapping = include_remote_device_mapping
+        self.module = AnsibleModule(argument_spec = dict())
+
+    def run(self):
+        """
+            Main method of the class
+        """
+        self.module.exit_json(ansible_facts = {
+            'console_facts' : self.get_console_facts()
+        })
+
+    def get_console_facts(self):
+        """
+        Retrieve console facts
+        """
+        facts = {
+            "enabled" : self.get_console_feature_status()
+        }
+
+        if facts["enabled"]:
+            facts["lines"] = self.get_console_lines_status()
+            if self.include_remote_device_mapping:
+                facts["remote_device_mapping"] = self.build_remote_device_mapping(facts["lines"])
+        return facts
+
+    def get_console_feature_status(self):
+        """
+        Retrieve console feature information
+        """
+        rt, out, err = self.module.run_command('sonic-db-cli CONFIG_DB HGET CONSOLE_SWITCH|console_mgmt enabled')
```

Comment:
When using earlier image such as 2018**, the module will fail due to the key is not exists.
Should we consider backward compatibility for this module? If so, we can avoid populate error here by always returning a False

Reply:
I will add a new fixture to filter out duts that versions are not greater or equal than 202012